public/app/features/scenes/scenes/cache.tsx

```diff
@@ -0,0 +1,149 @@
+import { Button } from '@grafana/ui';
+import React from 'react';
+import { SceneRadioToggle } from '../apps/SceneRadioToggle';
+import { Scene, SceneCanvasText, SceneSubMenu, SceneTimePicker } from '../components';
+import { SceneFlexLayout } from '../components/layout/SceneFlexLayout';
+import { EmbeddedScene } from '../components/Scene';
+import { SceneSubMenuSpacer } from '../components/SceneSubMenu';
+import { SceneObjectBase } from '../core/SceneObjectBase';
+import { SceneTimeRange } from '../core/SceneTimeRange';
+import { SceneComponentProps, SceneLayoutChildState } from '../core/types';
+import { VariableValueSelectors } from '../variables/components/VariableValueSelectors';
+import { SceneVariableSet } from '../variables/sets/SceneVariableSet';
+import { CustomVariable } from '../variables/variants/CustomVariable';
+import { QueryVariable } from '../variables/variants/query/QueryVariable';
+
+export function getCacheTest(standalone: boolean): Scene {
+  const getVariables = () =>
+    new SceneVariableSet({
+      variables: [
+        new CustomVariable({
+          name: 'metricCached',
+          cacheKey: 'metric',
```

Comment:
We maybe can generate the cacheKey automatically, and add options to configure the cache: no_cache, only_cache, cache_and_network, etc.

For example, for the variables, we know they always need a name, we can use the name as the key. Since the name "should be" unique, we should be safe with that. In case the name is not enough, we could generate a deterministic hash using different scene props.

But not exposing the cacheKey as a mandatory prop to have a cache by default will allow us to simplify the API, but I think is worth allowing them to change it if needed.

Reply:
@ivanortegaalba as @torkelo mentioned, the name has to be static in order to reuse other object's state.

Think about a query variable - we want to reuse it's state only, as reusing the entire scene object is tricky (rewiring dependencies). So in two scenes, we will have two objects defined, but sharing the same cache key for state reuse.